src/alias.ml

```diff
@@ -125,24 +126,45 @@ module Store = struct
     Format.fprintf fmt "@[<2>{@ alias@ =@ %a@ ;@ deps@ = (%a)@ }@]"
       pp entry.alias pp_deps entry.deps
 
-  type t = (Fq_name.t, entry) Hashtbl.t
+  type dir_status =
+    { mutable dir_aliases : entry list
+    ; mutable closed  : bool
```

Comment:
What's the use of this `closed` flag? I see it being set and asserted but it doesn't seem to affect behavior. Is there an invariant this is meant to maintain?

Reply:
There is indeed an invariant. I'll add a comment